atmos_dyamond_summer/create_artifact.jl

```diff
@@ -1,8 +1,276 @@
+    ncin = NCDataset(infile, "r")
+    ncout = NCDataset(outfile, "c")
+
+    lonidx = 1:skip:ncin.dim["lon"]
+    latidx = 1:skip:ncin.dim["lat"]
+    zidx = ncin.dim["lev"]:-1:1
+
+    nlon = length(lonidx)
+    nlat = length(latidx)
+    source_nz = length(zidx)
+
+    @inbounds begin
+        # get source z
+        sp = repeat(exp.(ncin["lnsp"][lonidx, latidx, 1, 1]), 1, 1, source_nz) # surface pressure
+        hyam = repeat(reshape(ncin["hyam"][zidx], 1, 1, source_nz), nlon, nlat, 1)
+        hybm = repeat(reshape(ncin["hybm"][zidx], 1, 1, source_nz), nlon, nlat, 1)
+
+        P = FT.(compute_P.(hyam, hybm, sp))
+        log_p_by_p0 = log.(P) .- log_P0
+        source_z = Plvl_inv2.(log_p_by_p0)
+
+        nz = source_nz #length(zidx)
+
+        # create a target z grid with nz points
+        target_z = FT.(Plvl_inv.(range(Plvl(z_min), Plvl(z_max), nz)))
+        # defining dimensions
+        defDim(ncout, "lon", nlon)
+        defDim(ncout, "lat", nlat)
+        defDim(ncout, "z", nz)
+        # longitude
+        lon = defVar(ncout, "lon", FT, ("lon",), attrib = ncin["lon"].attrib)
+        lon[:] = Array{FT}(ncin["lon"][lonidx])
+    
+        # latitude
+        lat = defVar(ncout, "lat", FT, ("lat",), attrib = ncin["lat"].attrib)
+        lat[:] = Array{FT}(ncin["lat"][latidx])
+
+        z = defVar(ncout, "z", FT, ( "z",), attrib = OrderedDict(
+            "Datatype" => string(FT),
+            "standard_name" => "altitude",
+            "long_name" => "altitude",
+            "units" => "m",
+            ),
+        )
+        z[:] = target_z
+
+        # p (pressure)
+        p = defVar(ncout, "p", FT, ("lon", "lat", "z",), attrib = OrderedDict(
+            "Datatype" => string(FT),
+            "standard_name" => "pressure",
+            "long_name" => "pressure",
+            "units" => "Pa",
+            ),
+        )
+        p[:, :, :] = P
+
+
+        # u (eastward_wind)
+        u = defVar(ncout, "u", FT, ("lon", "lat", "z",), attrib = ncin["u"].attrib)
+        u[:, :, :] = interpz_3d(target_z, source_z, ncin["u"][lonidx, latidx, zidx, 1])
+
+        # v (northward_wind)
+        v = defVar(ncout, "v", FT, ("lon", "lat", "z",), attrib = ncin["v"].attrib)
+        v[:, :, :] = interpz_3d(target_z, source_z, ncin["v"][lonidx, latidx, zidx, 1])
+    
+        # w (vertical velocity)
+        w = defVar(ncout, "w", FT, ("lon", "lat", "z",), attrib = ncin["w"].attrib)
+        w[:, :, :] = interpz_3d(target_z, source_z, ncin["w"][lonidx, latidx, zidx, 1])
+    
+        # t (air_temperature)
+        t = defVar(ncout, "t", FT, ("lon", "lat", "z",), attrib = ncin["t"].attrib)
+        t[:, :, :] = interpz_3d(target_z, source_z, ncin["t"][lonidx, latidx, zidx, 1])
+    
+        # q (specific_humidity)
+        q = defVar(ncout, "q", FT, ("lon", "lat", "z",), attrib = ncin["q"].attrib)
+        _q = interpz_3d(target_z, source_z, ncin["q"][lonidx, latidx, zidx, 1])
+        q[:, :, :] = max.(_q, FT(0))
```

Comment:
Are there negative values in `_q`?

Reply:
Yes. I noticed some small negative numbers.

Reply:
How about clwc, ciwc, crwc, and cswc then? We should limit them to 0 too if there are negative numbers.